src/Validator/HungarianIbanValidator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Rikudou\Iban\Validator;
+
+/**
+ * @see https://www.ecbs.org/Download/Tr201v3.9.pdf
+ */
+class HungarianIbanValidator implements ValidatorInterface
+{
+    private const WEIGHTS = [
+        9,
+        7,
+        3,
+        1,
+    ];
+
+    /** @var string */
+    private $accountNumber;
+
+    public function __construct(string $accountNumber)
+    {
+        $this->accountNumber = (string) preg_replace('/[\s\-]+/', '', $accountNumber);
+    }
+
```

Comment:
Tying to the comment above, this should be replaced to accept the object itself. I won't replace everything that should be fixed in this code review, you get the idea.

```suggestion
    private $iban;

    public function __construct(HungarianIbanAdapter $iban)
    {
        $this->iban = $iban;
    }

```